cmake/toolchain/xtools/target.cmake

```diff
@@ -10,7 +10,11 @@ set(CROSS_COMPILE_TARGET_arc         arc-zephyr-elf)
 set(CROSS_COMPILE_TARGET_x86      x86_64-zephyr-elf)
 set(CROSS_COMPILE_TARGET_sparc     sparc-zephyr-elf)
 
-set(CROSS_COMPILE_TARGET ${CROSS_COMPILE_TARGET_${ARCH}})
+if(DEFINED ENV{XTOOLS_CROSS_COMPILE_TARGET})
+  set(CROSS_COMPILE_TARGET "$ENV{XTOOLS_CROSS_COMPILE_TARGET}")
+else()
```

Comment:
your commit message says:
> When crosstool-ng is used as a build system, the target triple might not
include 'zephyr' -- arm-none-eabi is quite commonly used for compatibility
with the GNU Arm Embedded Toolkit.

but if you read the guide here: https://docs.zephyrproject.org/latest/develop/toolchains/crosstool_ng.html
> Follow the [Zephyr SDK with Crosstool NG instructions](https://github.com/zephyrproject-rtos/sdk-ng/blob/master/README.md) to build your toolchain. Repeat as necessary to build toolchains for multiple target architectures.

users are expected to use the Zephyr SDK build instructions for `xtools`.
If users are manually building using cross-tool they could even change more than just the triplet.
They could even set `CT_CC_GCC_CONFIG_TLS=n`.

Because of this, and because we already support a generic `CROSS_COMPILE` setting:
https://docs.zephyrproject.org/latest/develop/toolchains/other_x_compilers.html
Users with a different triplet can just do:
```
$ export ZEPHYR_TOOLCHAIN_VARIANT=cross-compile
$ export CROSS_COMPILE=/my/gcc/arm-none-eabi-
```

so I fail to see why we need to have more customization in `xtools` when they users can already accomplish exactly this, using `cross-compile`.
Let's not introduce more ways of achieving the same.

Especially since the `ZEPHYR_TOOLCHAIN_VARIANT=xtools` is a convenience that selects the exact toolchain based on the arch of the target, whereas the `XTOOLS_CROSS_COMPILE_TARGET` needs to be adjusted by the user when switching from `arm` to `xtensa`.
Thus the proposed solution in this commit is directly comparable to the use of `CROSS_COMPILE` where the arch is part of the triplet set by the user.

Reply:
Thanks for your explanation; I hadn't understood the difference between the xtools and cross-compile variants.

Reply:
And thanks for taking time to clean up this :+1: